forth_src/base.fs

```diff
@@ -106,8 +106,37 @@ then (to) ; immediate
 : 2drop ( a b -- )
 postpone drop postpone drop ; immediate
 
-: save-forth ( strptr strlen -- )
-801 here d word count saveb ;
+: :noname here 0 ] ;
+
+: dsize ( -- n) \ not really needed?
+top latest - ;
+
+: top! ( addr -- )
+latest swap dsize 2dup - to latest 
+over to top swap over - swap 1+ move ;
+
+\ first pass unfactored code ahead
+
+: compare ( caddr u caddr u -- 1 | 0 )
+2 pick over <> if 2drop 2drop 1 exit then
+drop swap
+begin ( caddr caddr u )
+1- 2dup + c@ ( caddr caddr u c )
+over 4 pick + c@ <> if ( caddr caddr u )
+2drop drop 1 exit then
+dup 0= if 2drop drop 0 exit then again ;
+
+: name>xta count $1f and + ;
+
+: (string>name) ( caddr u nt ) ( nt 0 | caddr u )
```

Comment:
These little lambda words used with `dowords` are probably reusable enough to be documented.